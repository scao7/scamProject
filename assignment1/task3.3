2 6.260000e-29
