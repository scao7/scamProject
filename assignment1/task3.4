2 6.530000e-29
