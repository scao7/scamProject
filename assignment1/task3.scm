(define (root-n degree)
  (define(rootNum number)
  (define tolerance 0.0000000000000001)
  (define(good-enough? next guess)
    (< (abs(- next guess)) tolerance)
    )
  (define (improve guess)
    (/ (+ (* (- degree 1) guess) (/ number (expt guess (- degree 1)))) degree)
  )
  (define (root-iter guess)
    (let ((next (improve guess)))
      (if (good-enough? next guess)
	guess
	(root-iter next))))
  (root-iter 1.0)
  )
 )




(define (main)
 (setPort (open (getElement ScamArgs 1) 'read))
  (define arg(readExpr))
  (define arg1(readExpr))
    ;(inspect ((root-n arg) arg1))
    (println "((root-n " arg ") " arg1 ") is " (fmt "%.15f" ((root-n arg)arg1)))
 )
