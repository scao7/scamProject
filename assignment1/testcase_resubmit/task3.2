2 5.970000e-29
