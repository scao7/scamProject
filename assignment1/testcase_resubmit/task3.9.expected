((root-n 10) 786) is 1.947790550656805
